### PR TITLE
New User List for Integration selector

### PR DESCRIPTION
**Pull request by javaguirre**
#### Summary

The plan is to separate the different data type lists into their own components, starting with Users.

1. Replace CustomList with ServerUserList for user selection, this will handle data processing and will pass data to the UserList component
2. Do the same with Channels
3. Do the same (kind of) with options
4. Fix multiselect to remove renders
5. Replace the new ServerUserList on CreateDirectMessage

Two multiselect state lists will coexist for a time while I replace the rest of the code. 

I haven't touched the UserList component as it's being worked on in a different PR, but I might need to do property changes when [the other PR gets merged](https://github.com/mattermost/mattermost-mobile/pull/6775).

I will create another PR for channel list selection with this one as the base so I can reuse code but not make it too long to review, and separate concerns.

@larkox let me know what you think about this approach. Thank you!

#### Checklist

- [X] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note
Replace user list render on integration selector
```

### CODE REVIEW

**Comment by javaguirre** [comment]
The channel list refactor is being done [here while this gets reviewed/merged](https://github.com/javaguirre/mattermost-mobile/pull/1)

File: app/screens/integration_selector/integration_selector.tsx [hunk 1]
```diff
@@ -288,10 +277,25 @@ function IntegrationSelector(
...
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            // New multiselect
+            handleSelect(Object.values(selectedIds) as UserProfile[]);
+        } else {
+            // Legacy multiselect
+            handleSelect(Object.values(multiselectSelected));
+        }
         close();
-    }, [multiselectSelected, handleSelect]);
+    }, [multiselectSelected, selectedIds, handleSelect]);
```
Comment (larkox) [inline app/screens/integration_selector/integration_selector.tsx:298]: Why not use the `multiselectSelected` for users like we do for channels and options? We can always form the selectedIds just by getting `Object.keys(multiselectSelected)`, right?

**Review by larkox — APPROVED** [review]
Great work, just one potential improvement.

File: app/screens/integration_selector/integration_selector.tsx (hunk 1 above)
Comment (javaguirre) [reply app/screens/integration_selector/integration_selector.tsx:298]: I'd like to remove it in the future so all of them use selectedIds and remove multiselectSelected from the code.

Comment (javaguirre) [reply app/screens/integration_selector/integration_selector.tsx:298]: I'll change it in the options pr 👌, I have the channel list pr ready too ☺️

**Pull request merged**